### PR TITLE
Add Log action for $password_reset_request.succeeded and $password_reset_request.failed events

### PR DESCRIPTION
**Pull request by dawlib**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -6,12 +6,17 @@ source "https://rubygems.org"
...
 gem "rake"
-gem "rspec"
-gem "rspec-rails"
-gem "simplecov"
 gem "standard"
 gem "sqlite3"
 gem "vcr"
-gem "webmock"
+
+group :development do
```
Comment (madejejej) [inline Gemfile:16]: we don't really need a separate group since it's a gem.

File: lib/castle_devise/patches/passwords_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      # POST /resource/password
+      def create
+        super do |resource|
+          next unless resource_class.castle_hooks[:after_password_reset_request]
+
+          CastleDevise.sdk_facade.log(
+            event: "$password_reset_requested",
+            status: resource.persisted? ? "$succeeded" : "$failed",
+            context: CastleDevise::Context.from_rack_env(request.env, scope_name, resource)
+          )
```
Comment (madejejej) [inline lib/castle_devise/patches/passwords_controller.rb:20]: Shouldn't we rescue errors here as well?

Comment (dawlib) [reply lib/castle_devise/patches/passwords_controller.rb]: Fixed

**Review by madejejej — APPROVED** [review]

File: Gemfile.lock [hunk 3]
```diff
@@ -150,10 +169,13 @@ GEM
       hashdiff (>= 0.4.0, < 2.0.0)
 
 PLATFORMS
+  ruby
```
Comment (opti) [inline Gemfile.lock:172]: @dawlib bumping up local rubygems version might help
```suggestion
```

**Review by madejejej — APPROVED** [review]

**Review by madejejej — APPROVED** [review]

**Pull request merged**